files/stunnel_build_test_certs.sh

```diff
@@ -0,0 +1,185 @@
+countryName             = optional
+stateOrProvinceName     = optional
+localityName            = optional
+organizationName        = optional
+organizationalUnitName  = optional
+commonName              = supplied
+subjectAltName          = optional
+[ req ]
+default_bits            = 2048
+default_keyfile         = privkey.pem
+distinguished_name      = req_distinguished_name
+attributes              = req_attributes
+x509_extensions = v3_ca                   # Extensions added to self-signed cert
+string_mask = utf8only
+[ usr_cert ]
+basicConstraints=CA:FALSE
+nsComment                       = "OpenSSL Generated Certificate"
+subjectKeyIdentifier=hash
+authorityKeyIdentifier=keyid,issuer
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+[ v3_ca ]
+subjectKeyIdentifier=hash
+authorityKeyIdentifier=keyid:always,issuer
+basicConstraints = CA:true
+[ crl_ext ]
+authorityKeyIdentifier=keyid:always
+[ proxy_cert_ext ]
+basicConstraints=CA:FALSE
+nsComment                       = "OpenSSL Generated Certificate"
+subjectKeyIdentifier=hash
+authorityKeyIdentifier=keyid,issuer
+proxyCertInfo=critical,language:id-ppl-anyLanguage,pathlen:3,policy:foo
+[ tsa ]
+default_tsa = tsa_config1                  # The default TSA section
+[ tsa_config1 ]
+dir             = ./demoCA                 # TSA root directory
+serial          = \$dir/tsaserial          # Current serial number (mandatory)
+crypto_device   = builtin                  # OpenSSL engine to use for signing
+signer_cert     = \$dir/tsacert.pem        # The TSA signing certificate
+                                           # (optional)
+certs           = \$dir/cacert.pem         # Cert. chain to include in reply
+                                           # (optional)
+signer_key      = \$dir/private/tsakey.pem # The TSA private key (optional)
+default_policy  = tsa_policy1              # Policy if request did'nt specify it
+                                           # (optional)
+other_policies  = tsa_policy2, tsa_policy3 # acceptable policies (optional)
+digests         = sha1                     # Acceptable message digests (req'd)
+accuracy        = secs:1, millisecs:500, microsecs:100  # (optional)
+clock_precision_digits  = 0                # number of digits after dot. (opt.)
+ordering                = yes              # Is ordering defined for timestamps?
+                                           # (optional, default: no)
+tsa_name                = yes              # Include the TSA name in the reply?
+                                           # (optional, default: no)
+ess_cert_id_chain       = no               # Include the ESS cert ID chain?
+                                           # (optional, default: no)
+EOF
+
+echo done
+
+echo
+echo "### Creating CA cert.  Enter pass phrase and Common Name when prompted."
+openssl req -new -x509 -days 3650 -extensions v3_ca -keyout cakey.pem -out cacert.pem -config stunnel-ssl.cnf
```

Comment:
FYI there is annoyance in the test clusters right now with the CA I created awhile back. Your script may fix up those issues. I'll point Allan here so he can try if he gets a chance.